Makefile

```diff
@@ -0,0 +1,61 @@
+.PHONY: help docker-start phpunit-full phpunit-mysql phpunit-mysql-without-dama phpunit-postgresql phpunit-postgresql-without-dama phpunit-mongo cs-fixer psalm
+
+MYSQL_URL="mysql://root:root@mysql:3306/zenstruck_foundry?charset=utf8"
+POSTGRESQL_URL="postgresql://root:root@postgres:5432/zenstruck_foundry?charset=utf8"
+MONGO_URL="mongodb://mongo:mongo@mongo:27017/mongo?compressors=disabled&amp;gssapiServiceName=mongodb"
+SQLITE_URL="sqlite://./var/app.db"
+
+INTERACTIVE:=$(shell [ -t 0 ] && echo 1)
+ifdef INTERACTIVE
+	DC_EXEC=docker-compose exec
+else
+	DC_EXEC=docker-compose exec -T
+endif
+
+DOCKER_PHP=${DC_EXEC} php
+
+.DEFAULT_GOAL := help
+
+help:
+	@fgrep -h "###" $(MAKEFILE_LIST) | fgrep -v fgrep | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+docker-start: ### create and run database containers
+	docker-compose up -d --build
+	docker-compose ps
+
+tests: cs-fixer psalm phpunit-mysql ### run full tests suite
+
+phpunit-full: vendor/autoload.php ### run full phpunit
+	@$(eval filter ?= '.')
+	docker-compose exec -e USE_FOUNDRY_BUNDLE=1 -e DATABASE_URL=${MYSQL_URL} -e MONGO_URL=${MONGO_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+
+phpunit-fast: ### run phpunit with only sqlite with DAMADoctrineTestBundle activated
+	@$(eval filter ?= '.')
+	${DC_EXEC} -e DATABASE_URL=${SQLITE_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+
+phpunit-mysql: vendor/autoload.php ### run phpunit with only mysql with DAMADoctrineTestBundle activated
+	@$(eval filter ?= '.')
+	${DC_EXEC} -e DATABASE_URL=${MYSQL_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+
+phpunit-postgresql: vendor/autoload.php ### run phpunit with only postgresgl with DAMADoctrineTestBundle activated
+	@$(eval filter ?= '.')
+	${DC_EXEC} -e DATABASE_URL=${POSTGRESQL_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+
+phpunit-mongo: vendor/autoload.php ### run phpunit with only mongo
+	@$(eval filter ?= '.')
+	${DC_EXEC} -e MONGO_URL=${MONGO_URL} php vendor/bin/simple-phpunit --filter=$(filter) --stop-on-failure
+
+psalm: bin/tools/psalm/vendor/vimeo/psalm/psalm ### run psalm
+	${DOCKER_PHP} bin/tools/psalm/vendor/vimeo/psalm/psalm --config=./psalm.xml || true
+
+bin/tools/psalm/vendor/vimeo/psalm/psalm: vendor/autoload.php
+	${DOCKER_PHP} composer bin psalm install
+
+cs-fixer: bin/tools/cs-fixer/vendor/friendsofphp/php-cs-fixer/php-cs-fixer ### run PHP CS-Fixer
+	${DOCKER_PHP} bin/tools/cs-fixer/vendor/friendsofphp/php-cs-fixer/php-cs-fixer --no-interaction --diff -v fix --config=.php-cs-fixer.dist.php
+
+bin/tools/cs-fixer/vendor/friendsofphp/php-cs-fixer/php-cs-fixer: vendor/autoload.php
+	${DOCKER_PHP} composer bin cs-fixer install
+
+vendor/autoload.php:
```

Comment:
Possible to auto-run `docker-start` if not already running?

Reply:
yes but... I need to find a way to detect if it is already running. Makefile syntax is a bit cumbersome